lib/SIL/GraphFunctionDeviceInfo.cpp

```diff
@@ -198,6 +218,17 @@ void GraphFunctionDeviceInfo::handleDevicePlacement(
        SymbolicValue::getString(deviceString, ctx.getAllocator())});
 }
 
+GraphFunctionDeviceInfo::GraphFunctionDeviceInfo(DeviceId primaryDeviceId,
+                                                 bool isTPUInfeedEnabled)
+    // When `TFUseDeviceStack` is true, ignore `primaryDeviceId` provided at
+    // compile time (e.g. via compiler flag); instead use runtime device info.
+    : primaryDeviceId(TFUseDeviceStack ? RuntimeDeviceId : primaryDeviceId),
+      isTPUInfeedEnabled(isTPUInfeedEnabled) {
+  assert(primaryDeviceId.type != DeviceType::ALL);
+  if (!TFUseDeviceStack)
```

Comment:
Use if (primaryDeviceId != RuntimeDeviceID) as it easier to understand?

Reply:
hmm that would seem to change the semantics. Here we want to say: in the old code path (when the flag is false), continue to add a primary device id upfront.

What we do for the new code path here is to TBD, and will be fleshed out in the coming PRs.

Reply:
I see. SG.